components/menu/menu-item-mixin.js

```diff
@@ -171,7 +180,12 @@ export const MenuItemMixin = superclass => class extends PropertyRequiredMixin(s
 		this.setAttribute('tabindex', '0');
 	}
 
-	__onKeyDown(e) {
+	_onHidden() {
+		/** Dispatched when the visibility of the menu item changes */
+		this.dispatchEvent(new CustomEvent('d2l-menu-item-visibility-change', { bubbles: true, composed: true }));
+	}
+
+	_onKeyDown(e) {
```

Comment:
We want to override this function for `d2l-menu-item-link`, otherwise the `d2l-menu-item-select` will be sent twice (once for `keydown` then again for `click`). We're renaming it to be more accurate and move the api in the direction we want (since this function is fairly specific to `d2l-menu-item`, and it's not surprising a consumer may need to override it.